pysteps/extrapolation/semilagrangian.py

```diff
@@ -113,6 +116,15 @@ def extrapolate(
     n_iter = kwargs.get("n_iter", 1)
     return_displacement = kwargs.get("return_displacement", False)
 
+    # replace nans with zeros and apply separate masking if interp_order > 1
+    if interp_order > 1:
+        mask = np.isfinite(precip)
+        precip = precip.copy()
+        precip[~mask] = 0.0
+        mask = mask.astype(float)
+
+    prefilter = True if interp_order > 1 else False
```

Comment:
could you please briefly explain what the prefilter does? should we mention it in the docstrings?

Reply:
I'm not sure if there is any need to mention this in the docstrings. As stated in the documentation of `scipy.ndimage.interpolation.map_coordinates`, prefiltering is necessary when order>1. However, the details are not explained. I tested order=3 without prefiltering. Compared to order=1, this produced a very smooth output with even more loss of spectral power.